mosaic-runtime/src/commonTest/kotlin/com/jakewharton/mosaic/layout/ParentDataModifierTest.kt

```diff
@@ -0,0 +1,134 @@
+package com.jakewharton.mosaic.layout
+
+import androidx.compose.runtime.Composable
+import assertk.assertThat
+import assertk.assertions.hasSize
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import com.jakewharton.mosaic.AtLeastSize
+import com.jakewharton.mosaic.Holder
+import com.jakewharton.mosaic.modifier.Modifier
+import com.jakewharton.mosaic.testing.runMosaicTest
+import com.jakewharton.mosaic.ui.Color
+import com.jakewharton.mosaic.ui.Layout
+import com.jakewharton.mosaic.ui.unit.Constraints
+import kotlin.test.Test
+import kotlinx.coroutines.test.runTest
+
+class ParentDataModifierTest {
+	// Test that parent data defaults to null
+	@Test fun parentDataDefaultsToNull() = runTest {
+		runMosaicTest {
+			val parentData = Holder<Any?>(null)
+			setContent {
+				Layout(
+					content = { SimpleDrawChild() },
+					measurePolicy = { measurables, constraints ->
+						assertThat(measurables).hasSize(1)
+						parentData.value = measurables[0].parentData
+
+						val placeable = measurables[0].measure(constraints)
+						layout(placeable.width, placeable.height) { placeable.place(0, 0) }
+					},
+				)
+			}
+			awaitSnapshot()
+			assertThat(parentData.value).isNull()
+		}
+	}
+
+	// Test that parent data doesn't flow to grandchild measurables. They must be
+	// reset on every Layout level
+	@Test fun parentDataIsReset() = runTest {
+		runMosaicTest {
+			val parentData = Holder<Any?>(null)
+			setContent {
+				Layout(
+					modifier = Modifier.layoutId("Hello"),
+					content = { SimpleDrawChild() },
+					measurePolicy = { measurables, constraints ->
+						assertThat(measurables).hasSize(1)
+						parentData.value = measurables[0].parentData
+
+						val placeable = measurables[0].measure(constraints)
+						layout(placeable.width, placeable.height) { placeable.place(0, 0) }
+					},
+				)
+			}
+			awaitSnapshot()
+			assertThat(parentData.value).isNull()
+		}
+	}
+
+	@Test fun multiChildLayoutTest_doesNotOverrideChildrenParentData() = runTest {
+		runMosaicTest {
+			setContent {
+				val header =
+					@Composable {
+						Layout(modifier = Modifier.layoutId(0), content = {}) { _, _ ->
+							layout(0, 0) {}
+						}
+					}
+				val footer =
+					@Composable {
+						Layout(modifier = Modifier.layoutId(1), content = {}) { _, _ ->
+							layout(0, 0) {}
+						}
+					}
+
+				Layout({
+					header()
+					footer()
+				}) { measurables, _ ->
+					assertThat((measurables[0].parentData as? LayoutIdParentData)?.layoutId).isEqualTo(0)
+					assertThat((measurables[1].parentData as? LayoutIdParentData)?.layoutId).isEqualTo(1)
```

Comment:
Any reason to avoid `layoutId` helper extension here?

Reply:
I wanted to write code to explicitly get the value from `parentData` without hiding it behind an abstraction, even a simple one.